src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetRuleSubscriberBase.cs

```diff
@@ -243,10 +253,18 @@ private IEnumerable<string> GetWatchedRules(RuleHandlerType handlerType)
                                                   isActiveContext,
                                                   ruleChangeContext)
                                      .ConfigureAwait(true);
+
+                        // check full set of handler rules for changes
+                        _treeTelemetryService.ObserveHandlerRulesChanges(
+                            projectContextToUpdate.TargetFramework,
+                            handler.GetRuleNames(RuleHandlerType.DesignTimeBuild), 
```

Comment:
>handler [](start = 28, length = 7)

What if handler does not support DesignTimeBuild type? In this case you would create an observable for it but it would never have changes and thus telemetry event would never be sent? It could be legitimate case when handler has only evaluation, but you would either not send event at all or sent it as Unresolved only.

Reply:
Resolved this by checking if handler supports DesignTime build

---
In reply to: [145182121](https://github.com/dotnet/project-system/pull/2879#discussion_r145182121) [](ancestors = 145182121)